src/wp-includes/block-template-utils.php

```diff
@@ -37,21 +37,8 @@
  * }
  */
 function get_block_theme_folders( $theme_stylesheet = null ) {
-	$theme_name = null === $theme_stylesheet ? get_stylesheet() : $theme_stylesheet;
-	$root_dir   = get_theme_root( $theme_name );
-	$theme_dir  = "$root_dir/$theme_name";
-
-	if ( file_exists( $theme_dir . '/block-templates' ) || file_exists( $theme_dir . '/block-template-parts' ) ) {
-		return array(
-			'wp_template'      => 'block-templates',
-			'wp_template_part' => 'block-template-parts',
-		);
-	}
-
-	return array(
-		'wp_template'      => 'templates',
-		'wp_template_part' => 'parts',
-	);
+	$theme = wp_get_theme( $theme_stylesheet );
```

Comment:
If `$theme_stylesheet` is not passed, this will pass `null` to `wp_get_theme()`, which expects a string.

Reply:
We also need to check if the theme exists. What happens if you put a stylesheet that is invalid into the function.

Reply:
@costdev A nice point. I have update the code to handle it. 

@spacedmonkey  I feel we don't need to add check for theme exists. `wp_get_theme`  returns an instance of `WP_Theme` irrespective of theme's existence and (new WP_Theme)->get_block_template_folders() do returns a default value which aligns to the current default of `get_block_theme_folders`. The cache too are updated in a way to respect these defaults.

Reply:
I disagree, there is no problem with adding this check is careful programming. As this a developer function and developers pass an invalid theme name.

Reply:
@spacedmonkey I agree with you here on careful programming. I was saying we don't need check here as `WP_Theme` already does this check inside `__construct` and we do set defaults for `$block_template_folders` there.

By not calling `exists()` we can avoid a `in_array` check. 

ref: 
https://github.com/WordPress/wordpress-develop/pull/5202/files#diff-394d564a692a6efc5195fef20d41cb1c7bcdbc3862325f312433ace9856d2f6fR284

https://github.com/WordPress/wordpress-develop/pull/5202/files#diff-394d564a692a6efc5195fef20d41cb1c7bcdbc3862325f312433ace9856d2f6fL306

Reply:
Noting the documented return value of `wp_get_theme()`:
> Theme object. Be sure to check the object's exists() method if you need to confirm the theme's existence.
> [Reference](https://github.com/WordPress/wordpress-develop/blob/6.3/src/wp-includes/theme.php#L114)

A call to [WP_Theme::exists()](https://github.com/WordPress/wordpress-develop/blob/6.3/src/wp-includes/class-wp-theme.php#L729-L731) will check the `$errors` property first.

Looking at [this condition](https://github.com/WordPress/wordpress-develop/pull/5202/files#diff-394d564a692a6efc5195fef20d41cb1c7bcdbc3862325f312433ace9856d2f6fR284), note that this sets `$this->errors`, so the `in_array()` will be skipped.

Looking at [this condition](https://github.com/WordPress/wordpress-develop/pull/5202/files#diff-394d564a692a6efc5195fef20d41cb1c7bcdbc3862325f312433ace9856d2f6fL306), note that this also sets `$this->errors`, so the `in_array()` will also be skipped.

Genuine question: Is there a case for getting the default `$block_template_folders` value even if the theme doesn't exist or isn't readable?

Reply:
Whether there's a case or not, I suspect that this wasn't taken into account before, and that protecting BC means ignoring whether the theme exists.

It seems like we'd benefit greatly from adding tests for `get_block_theme_folders()` to Core before continuing with this change.

Reply:
@costdev I am not aware of a case where we would be needing templates if the theme doesn't exists.

However, the current trunk does support default irrespective of themes status, leading us to support default.

Reply:
The way that the code currently works is technically correct as far as I can tell. The function should continue to return the default block theme folders even if the theme doesn't exist - the same way it does so far.

At the same time, I'm not opposed to adding a check for `WP_Theme::exists()`, just to make the code more intuitive, which can help avoid future errors.